v2/tools/generator/internal/astmodel/type_visitor_builder.go

```diff
@@ -426,5 +431,14 @@ func (b *TypeVisitorBuilder[C]) buildVisitInterfaceType() func(*TypeVisitor[C],
 		}
 	}
 
-	panic(fmt.Sprintf("unexpected InterfaceType func %#v", b.VisitInterfaceType))
+	panic(fmt.Sprintf("unexpected InterfaceType func %#v (%s)", b.VisitInterfaceType, b.diagnosticClue()))
+}
+
+func (b *TypeVisitorBuilder[C]) diagnosticClue() string {
```

Comment:
Its not immediately clear to me why `Caller(3)` in this function. Might help if it had a comment at the function level explaining what this function does/what its expected output is, and possibly a comment on the `Caller(3)` explaining whats special about 3.

Reply:
I've added a detailed comment to explain.